modules/python/src2/cv2.cpp

```diff
@@ -392,6 +392,7 @@ static bool pyopencv_to(PyObject* o, Mat& m, const ArgInfo& info)
             oarr = (PyArrayObject*) o;
         }
         else {
+            CV_Assert(PyArray_ISCONTIGUOUS(oarr));
             oarr = PyArray_GETCONTIGUOUS(oarr);
```

Comment:
According to https://docs.scipy.org/doc/numpy-1.13.0/reference/c-api.array.html#c.PyArray_GETCONTIGUOUS it looks like `PyArray_GETCONTIGUOUS` should manage both cases.